File: mario.c
#include <cs50.h>
#include <stdio.h>

int get_num(void);

int main(void)
{
    int n = get_num();
        
    //rows
    for (int i = 1; i <= n; i++)
    {
        //cols       
        //spaces: total number of rows (n) aka the height minus the actual row number(i)
       for (int j = 1; j <= (n - i); j++)
       {
            printf(" ");
       }
       for (int k = 1; k <= i; k++)
       {
            printf("#");
       }
        printf("\n");
    }     
}


//Prompt user for number if number is less than 1 or greater than 8 reprompt
int get_num(void)
{
    int num;
    do 
    {
        num = get_int("Height: Pick a number 1-8 inclusive: \n");  
    }
    while (num < 1 || num > 8);
    return num;
}

File: mario.py
import sys
from cs50 import get_int


def main():
    # Call Prompt for User Input
    height = prompt()
    # Run Pyramid if True else Exit
    if height:
        pyramid(height)
    else:
        sys.exit()

# Gather Input from User


def prompt():
    # Run until User Provides a Number between 1-8
    while True:
        height = get_int("Height: ")
        if height > 0 and height < 9:
            break
    return height

# Print Pyramid


def pyramid(height):
    # Loop up to height
    for i in range(height):
        # Spaces are the Difference Between Total Number of Rows and the Row Number
        print(" " * (height - i - 1), end="")
        # Hashes are One More then the Row Number
        print("#" * (i + 1), end="")
        print()
        

if __name__ == "__main__":
    main()